src/sentry/sentry_metrics/indexer/strings.py

```diff
@@ -167,30 +170,42 @@ def __init__(self, indexer: StringIndexer) -> None:
     def bulk_record(
         self, use_case_id: UseCaseKey, org_strings: Mapping[int, Set[str]]
     ) -> KeyResults:
-        static_keys = KeyCollection(org_strings)
-        static_key_results = KeyResults()
-        for org_id, string in static_keys.as_tuples():
+        res = self._uca_bulk_record({REVERSE_METRIC_PATH_MAPPING[use_case_id]: org_strings})
+        return res.results[REVERSE_METRIC_PATH_MAPPING[use_case_id]]
+
+    def record(self, use_case_id: UseCaseKey, org_id: int, string: str) -> Optional[int]:
+        return self._uca_record(REVERSE_METRIC_PATH_MAPPING[use_case_id], org_id, string)
+
+    def _uca_bulk_record(
+        self, strings: Mapping[UseCaseID, Mapping[OrgId, Set[str]]]
+    ) -> UseCaseKeyResults:
+        static_keys = UseCaseKeyCollection(strings)
+        static_key_results = UseCaseKeyResults()
+        for use_case_id, org_id, string in static_keys.as_tuples():
             if string in SHARED_STRINGS:
                 id = SHARED_STRINGS[string]
-                static_key_results.add_key_result(
-                    KeyResult(org_id, string, id), FetchType.HARDCODED
+                static_key_results.add_use_case_key_result(
+                    UseCaseKeyResult(use_case_id, org_id, string, id), FetchType.HARDCODED
                 )
 
-        org_strings_left = static_key_results.get_unmapped_keys(static_keys)
+        org_strings_left = static_key_results.get_unmapped_use_case_keys(static_keys)
```

Comment:
kind of a nit, but could we rename the `org_strings_left` to something else that indicates that these are use case-aware strings?

Reply:
I feel like the name is pretty descriptive and we don't really have another `org_strings_left` that is not use case aware.

Reply:
Sure, I was thinking of distinguishing it from the old org_strings_left that we already had in this code. Also I'm thinking of how we renamed things like `use_case_results` etc for the other indexers? 

Also it could potentially make this line of code clearer, and truly indicate that it contains use case information.
`for use_case_id, key_collection in org_strings_left.mapping.items()`